File: Markov_Matrix/Markov_Matrix.py
'''
Markov Matrix is a matrix in which the sum of each row is equal to 1.
'''

def markovMatrix(matrix, row, col):       
    count = 0 
    for i in range(0, row):           
        # Calculating sum of each row
        sum1 = 0
        for j in range(0, col): 
            sum1 = sum1 + matrix[i][j] 
        if (sum1 != 1):
            count = count + 1
    if (count == 0):
        print("YES")
    else: 
        print("NO") 

# Driver program 
row = int(input())
col = int(input())
matrix = []
for i in range(row):  
    a = [] 
    for j in range(col):     
        a.append(float(input())) 
    matrix.append(a) 
markovMatrix(matrix, row, col) 

'''
Input:
row = 2
col = 2
Matrix = {{1, 0},
          {0.5, 0.5}}
Output:
YES
'''

File: Markov_Matrix/Markov_Matrix.cpp
/*
Markov Matrix is a matrix in which the sum of each row is equal to 1.
*/

#include<iostream> 
using namespace std; 

const int x = 10; 

void markovMatrix(double matrix[][x], int row, int col) 
{ 
    int count = 0;
    for (int i = 0; i < row; i++)
    { 
        double sum = 0;
        
        // Calculating sum of each row
        for (int j = 0; j < col; j++) 
            sum = sum + matrix[i][j]; 
        if (sum != 1) 
            count++; 
    } 
    if (count == 0)
        cout<<"YES";
    else 
        cout<<"NO";
} 

// Driver Code 
int main() 
{ 
    int row, col;
    double matrix[x][x];
    cin>>row;
    cin>>col;
    for (int i = 0; i < row; i++)
        for (int j = 0; j < col; j++)
            cin>>matrix[i][j]; 
    markovMatrix(matrix, row, col); 
} 

/*
Input:
row = 2
col = 2
Matrix = {{1, 0},
          {0.5, 0.5}}
Output:
YES
*/
